tempesta_fw/t/unit/kallsyms_helper.c

```diff
@@ -0,0 +1,53 @@
+/**
+ *              Tempesta FW
+ *
+ * Copyright (C) 2015 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <linux/string.h>
+#include <linux/kallsyms.h>
+#include <linux/module.h>
+
+#include "kallsyms_helper.h"
+
+typedef struct {
+        unsigned long addr;
+        const char    *name;
+} Symdata;
+
+static int
+get_sym(void *data, const char *namebuf, struct module *owner, unsigned long addr)
+{
+        Symdata *symdata = (Symdata *)data;
```

Comment:
no explicit casting is required here, `(void *)` is implicitly converted to any other pointer.